### PR TITLE
Parameterize withApiServices / withCustomTestDatabaseInternal

### PR DESCRIPTION
**Pull request by jmthibault79**
Solves the mysterious problem of the always-passing test.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Make sure liquibase is updated if appropriate
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it **(apply requires_doge label)**
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed
- some tests were not running: see GAWB-789

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiServiceSpec.scala [hunk 1]
```diff
@@ -21,7 +21,7 @@ class AdminApiServiceSpec extends ApiServiceSpec {
 
   case class TestApiService(dataSource: SlickDataSource, gcsDAO: MockGoogleServicesDAO)(implicit val executionContext: ExecutionContext) extends ApiServices with MockUserInfoDirectives
 
-  def withApiServices(dataSource: SlickDataSource)(testCode: TestApiService => Any): Unit = {
+  def withApiServices[T](dataSource: SlickDataSource)(testCode: TestApiService =>  T): T = {
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiServiceSpec.scala:24]: I would really like to refactor the duplicated code here, but it's nontrivial so I'm leaving that for another day: https://broadinstitute.atlassian.net/browse/GAWB-790

**Comment by jmthibault79** [comment]
I'm confident that I have solved this particular problem, but other problems may lurk related to `Unit` return types: https://broadinstitute.atlassian.net/browse/GAWB-789

**Comment by helgridly** [comment]
Can you explain how the lack of type param was causing the problem?

**Comment by jmthibault79** [comment]
`withCustomTestDatabaseInternal` was not evaluating its `testCode` function (returning type T) in some cases because the function itself (type (_ => T)) was also an acceptable thing to return.

**Comment by davidangb** [comment]
👍 LGWG. Tests pass _and_ fail appropriately for me on this branch.

**Comment by dvoet** [comment]
👍

**Pull request merged**